extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzBuildTimeConfig.java

```diff
@@ -78,4 +84,27 @@ public class QuartzBuildTimeConfig {
     @ConfigDocMapKey("plugin-name")
     @ConfigDocSection
     public Map<String, QuartzExtensionPointConfig> plugins;
+
+    /**
+     * Custom job store.
+     * <p>
+     * For instance,
+     * <ul>
+     * <li>quarkus.quartz.custom-store.class=org.acme.quartz.MyCustomJobStore
+     * <li>quarkus.quartz.custom-store.properties.myProp1=MyPropValue1
+     * <li>quarkus.quartz.custom-store.properties.myProp2=MyPropValue2
+     * </ul>
+     */
+    @ConfigItem
+    @ConfigDocMapKey("custom-store")
+    @ConfigDocSection
+    protected QuartzExtensionPointOptionalConfig customStore;
```

Comment:
Wouldn't `Optional<QuartzExtensionPointConfig> customStore` work as well?

Reply:
Pull branch 'main',  tried but still error

Error:
```txt
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Can not set java.util.Map field io.quarkus.quartz.runtime.QuartzExtensionPointConfig.properties to java.util.Optional
```

Log:
```java
[INFO] Running io.quarkus.quartz.test.subclass.CustomStoreTypeTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.208 s <<< FAILURE! - in io.quarkus.quartz.test.subclass.CustomStoreTypeTest
[ERROR] io.quarkus.quartz.test.subclass.CustomStoreTypeTest  Time elapsed: 0.208 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalArgumentException: Can not set java.util.Map field io.quarkus.quartz.runtime.QuartzExtensionPointConfig.properties to java.util.Optional
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:683)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$10(ClassBasedTestDescriptor.java:381)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:381)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:205)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:80)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.lang.RuntimeException: java.lang.IllegalArgumentException: Can not set java.util.Map field io.quarkus.quartz.runtime.QuartzExtensionPointConfig.properties to java.util.Optional
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:633)
	... 42 more
Caused by: java.lang.IllegalArgumentException: Can not set java.util.Map field io.quarkus.quartz.runtime.QuartzExtensionPointConfig.properties to java.util.Optional
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:167)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.throwSetIllegalArgumentException(UnsafeFieldAccessorImpl.java:171)
	at java.base/jdk.internal.reflect.UnsafeFieldAccessorImpl.ensureObj(UnsafeFieldAccessorImpl.java:58)
	at java.base/jdk.internal.reflect.UnsafeObjectFieldAccessorImpl.get(UnsafeObjectFieldAccessorImpl.java:36)
	at java.base/java.lang.reflect.Field.get(Field.java:425)
	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.getFieldAsMap(BuildTimeConfigurationReader.java:682)
	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.getMap(BuildTimeConfigurationReader.java:633)
	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader$ReadOperation.run(BuildTimeConfigurationReader.java:433)
	at io.quarkus.deployment.configuration.BuildTimeConfigurationReader.readConfiguration(BuildTimeConfigurationReader.java:322)
	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:187)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:104)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:328)
	... 44 more
``

Reply:
Interesting. It seems that it should work according to the docs: https://quarkus.io/guides/writing-extensions#optional-configuration-groups 

CC @radcortez am I missing something ^?

Reply:
Depends. There are some edge cases that currently don't work with @ConfigRoot, but do work with @ConfigMapping. From what I can tell from the stacktrace, it seems that the issue is a Map inside the Optional group. I'll have to check if this is supported with @ConfigRoot, but I think not.